pkg/engine/lifecycletest/alias_test.go

```diff
@@ -1648,3 +1648,102 @@ func TestSplitUpdateComponentAliases(t *testing.T) {
 	assert.Equal(t, resource.URN("urn:pulumi:test::test::pkgA:m:typB$pkgA:m:typC::resC"), snap.Resources[2].URN)
 	assert.Equal(t, resource.URN("urn:pulumi:test::test::pkgA:m:typB::resB"), snap.Resources[2].Parent)
 }
+
+func TestFailDeleteDuplicateAliases(t *testing.T) {
+	t.Parallel()
+
+	// This is a test for https://github.com/pulumi/pulumi/issues/14041 to check that we don't courrupt state files when
+	// old aliased resources are left in state because they can't delete.
```

Comment:
Could you add a detailed description of the bug here or on that issue (basically what you dumped in Slack)?